src/helpers/accessiblity.ts

```diff
@@ -233,3 +233,34 @@ export function isElementSelected(
   const { accessibilityState, 'aria-selected': ariaSelected } = element.props;
   return ariaSelected ?? accessibilityState?.selected ?? false;
 }
+
+export function getAccessibleName(
+  element: ReactTestInstance
+): string | undefined {
+  const labelTextFromLabel = getAccessibilityLabel(element);
+  const labelTextFromLabelledBy = getAccessibilityLabelledBy(element);
+
+  if (labelTextFromLabelledBy) {
+    const parentElement = getHostParent(element);
```

Comment:
labelled by does not have to be necessarily nested under the same parent. You should look from root element. simllar to: https://github.com/callstack/react-native-testing-library/blob/a9f32a69e46bedc9413dd0bdf74637f7a3b2b579/src/helpers/matchers/matchLabelText.ts#L40

You can access current root with `screen.UNSAFE_root`